### PR TITLE
MQMQA phase check and levelling allocation fix

### LINKED ISSUE #141: MQMQA pair outputs for API
**Reason.**
Quads are the "species" in MQMQA and as such their values are returned when looking for mole fractions for example. However, users might be more interested in the pairs which are more obvious as species for multiphysics applications. Let the users request pairs are outputs

**Proposed Solution**
Not much on Thermochimica side. Just add a function to check if a phase is MQMQA phase, a function to return the possible pairs in addition to quads. The other functions are already in `MQMQAFunctions.f90`.

**Impact**
User facing interface improves.

**Additional context**
Will be required by a MOOSE Thermochimica Integration PR.

### PR DESCRIPTION
**Pull request by parikshitbajpai**
Closes #141, #143

### CODE REVIEW

**Comment by parikshitbajpai** [comment]
I will try to add utility functions for checking pair in the constructor later

File: src/api/checkUnits.C [hunk 1]
```diff
@@ -1,112 +1,112 @@
...
-    return _element_names[atomic_number-1];
-  else
-    mooseError("elementName: Requesting an element with atomic number ", atomic_number, ", which is outside of the known range.");
-}
+    try
+    {
+      if (atomic_number >= 1 && atomic_number <= _element_names.size())
+        return _element_names[atomic_number - 1];
+      else
+        throw(atomic_number);
```
Comment (dschwen) [inline src/api/checkUnits.C:104]:
```suggestion
        throw std::range_error("elementName: Requesting an element with atomic number " + std::to_string(atomic_number) + ", which is outside of the known range.");
```

File: src/api/checkUnits.C [hunk 2]
```diff
@@ -1,112 +1,112 @@
...
+    }
+    catch (...)
+    {
+      std::cout << "elementName: Requesting an element with atomic number " << atomic_number << ", which is outside of the known range.";
+      return std::string();
+    }
```
Comment (dschwen) [inline src/api/checkUnits.C:110]:
```suggestion
```

File: src/api/checkUnits.C [hunk 3]
```diff
@@ -1,112 +1,112 @@
...
+      if (it != _atomic_number_map.end())
+        return it->second;
+      else
+        throw(element);
+    }
+    catch (...)
+    {
+      std ::cout << "atomicNumber: unknown chemical element " << element << " requested. Typo?";
+      return 0;
+    }
```
Comment (dschwen) [inline src/api/checkUnits.C:79]: see below

File: src/api/checkUnits.C [hunk 4]
```diff
@@ -1,112 +1,112 @@
...
+      if (std::find(_mass_names.begin(), _mass_names.end(), mass_unit) == _mass_names.end())
+        throw(mass_unit);
     }
-  else
-    mooseError("checkMass: unknown mass unit ", mass_unit, ". Typo?");
-}
+    catch (...)
+    {
+      std::cout << "checkMass: unknown mass unit " << mass_unit << ". Typo?";
+    }
```
Comment (dschwen) [inline src/api/checkUnits.C:58]: see below

Comment (dschwen) [reply src/api/checkUnits.C:58]: and fix all the try catch cases above

File: src/api/checkUnits.C (hunk 2 above)
Comment (dschwen) [inline src/api/checkUnits.C]:
```suggestion
    if (atomic_number >= 1 && atomic_number <= _element_names.size())
      return _element_names[atomic_number - 1];
    else
      throw std::range_error("elementName: Requesting an element with atomic number " + std::to_string(atomic_number) + ", which is outside of the known range.");
```

**Review by dschwen — APPROVED** [review]

**Pull request merged**